packages/auth/src/grant/routes.ts

```diff
@@ -96,14 +87,38 @@ async function createGrantInitiation(
 
   const { body } = ctx.request
   const { grantService, config } = deps
-  if (!validateGrantRequest(body)) {
+
+  if (
+    !deps.config.incomingPaymentInteraction &&
+    body.access_token.access
+      .map((acc) => {
+        return isIncomingPaymentAccessRequest(acc as IncomingPaymentRequest)
+      })
+      .every((el) => el === true)
+  ) {
+    const grant = await grantService.create(body)
+    const accessToken = await deps.accessTokenService.create(grant.id)
```

Comment:
Should these happen in a transaction?

Reply:
I don't think so. If grant creation fails, we'll never get to the token creation. And if token creation fails, the grant should still be granted. Or am I misunderstanding this?

Reply:
I think the main benefit would be rolling back grant creation if access token creation fails.
>And if token creation fails, the grant should still be granted. 

As is, I think we'd 500 and the request would need to be retried.

Reply:
When you put it this way, I guess you are right.

Reply:
But I get a foreign key error because the transaction to create the grant has not been committed and hence I can't insert the access token with the grant id foreign key. 🤔

Reply:
This appears to work: https://github.com/wilsonianb/rafiki/pull/134/commits/05b300dc0c176eeece8cf76d86543d4e7d17e8fe?diff=unified&w=1